### PR TITLE
Kuba/ssh/receive exit fix options construct fix/otp 19246/otp 19247

### PR DESCRIPTION
**Pull request by u3s**
Fix for GH-8223 and also append `{active, false}` at the end of socket options.

Server supervision tree improvements are planned to be done in other PR.

### CODE REVIEW

**Review by Maria-12648430 — APPROVED** [review]

File: lib/ssh/src/ssh.erl [hunk 1]
```diff
@@ -355,7 +355,7 @@ connect(Host0, Port, UserOptions, NegotiationTimeout)
             {error, Reason};
 
         Options ->
-            SocketOpts = [{active,false} | ?GET_OPT(socket_options,Options)],
```
Comment (IngelaAndin) [inline lib/ssh/src/ssh.erl:358]: I do not understand, we want the socket option to for sure be {active,false} and the last value added wins (proplist behavior and commandline option behaviour)

>proplists:get_value(active, [{active, false}, {active, true}]).
false

Also probably handle_options is suppose to no include the active option for socket options as it should be emulated.

File: lib/ssh/src/ssh_tcpip_forward_acceptor.erl [hunk 2]
```diff
@@ -36,9 +36,8 @@ supervised_start(FwdSup, {ListenAddrStr, ListenPort}, ConnectToAddr, ChanType, C
     case get_fwd_listen_opts(ListenAddrStr) of
         {ok,Opts} ->
             %% start listening on Addr:BoundPort
-            case gen_tcp:listen(ListenPort, [binary,
```
Comment (IngelaAndin) [inline lib/ssh/src/ssh_tcpip_forward_acceptor.erl:39]: Same as above I do not see the point. Just causing unnecessary overhead.

File: lib/ssh/src/ssh.erl (hunk 1 above)
Comment (u3s) [reply lib/ssh/src/ssh.erl:358]:
```
1> f(), {ok, ListenSocket} = gen_tcp:listen(8091, [{active,true}, binary, {active, false}]), inet:getopts(ListenSocket, [active]).
{ok,[{active,false}]}
```

Comment (IngelaAndin) [reply lib/ssh/src/ssh.erl:358]: That was surprising!

Comment (IngelaAndin) [reply lib/ssh/src/ssh.erl:358]: Or maybe actually the surprising thing is that you could get socket opts here that could already contain  active settings.
Emulated options should not be settable by the user. But as listen works like that you are correct it has to be appended to override the user option if provided.  This however is different form how many of our API functions work as they are perceived as having proplists semantics even if they are handled as proplists internally.

Comment (IngelaAndin) [reply lib/ssh/src/ssh.erl:358]: But yes you are right ssl:listen behaves the same for the socket options.  Which is not that surprising :) But I have this very strong memory of other options having to behave as proplist.  So sure ssl does Options ++ internal_inet_values() so I guess I changed my mind :)

Comment (u3s) [reply lib/ssh/src/ssh.erl:358]: ssl does it in similar way appends active option at the end of the list

```
lib/ssl/src/tls_socket.erl-listen(Transport, Port, #config{transport_info = {Transport, _, _, _, _},
lib/ssl/src/tls_socket.erl-                             inet_user = Options,
lib/ssl/src/tls_socket.erl-                             ssl = SslOpts, emulated = EmOpts} = Config) ->
lib/ssl/src/tls_socket.erl:    case Transport:listen(Port, Options ++ internal_inet_values()) of
lib/ssl/src/tls_socket.erl-     {ok, ListenSocket} ->
```

Comment (IngelaAndin) [reply lib/ssh/src/ssh.erl:358]: @u3s I think we wrote the last comments somewhat at the same time ...

Comment (IngelaAndin) [reply lib/ssh/src/ssh.erl:358]: Maybe change the commit message to not say proplist, as it does not behave as a proplist. Coule use option list instead.

**Review by IngelaAndin — APPROVED** [review]

**Pull request merged**